### ISSUE TITLE
GRIFFIN Restructure Issues

### ISSUE THREAD

**Issue by scshadrick** [GRIFFINCollaboration/detectorSimulations_v10#63]
When I move a GRIFFIN detector, the shell itself moves, however, the coloured crystals on the inside maintain their current position.  How do we change the positions of the the crystals as well?  

Cheers.

![image](https://user-images.githubusercontent.com/20971746/49023383-52976300-f154-11e8-88f6-1555c609aefa.png)

**Comment by VinzenzBildstein** [comment]
After looking a bit into this, my guess is that the way the coordinates for the crystals are calculated is done wrong if the coordinates are not the normal GRIFFIN ones. At least I can verify that changing the coordinates of the moveGermanium vector does change where the crystals are placed.
To test this I've added a z-offset of 100 (whatever units) for the first (blue crystal):
![griffinbluezplus100](https://user-images.githubusercontent.com/7277216/49825650-ee2ef300-fd52-11e8-860b-048ec71b0b4b.png)

The coordinates are calculated using the TransX, TransY, and TransZ functions of DetectionSystemGriffin with x = +/- (fGermaniumWidth + fGermaniumSeparation)/2.0, y = +/- (fGermaniumWidth + fGermaniumSeparation)/2.0, z = fGermaniumLength/2.0 + fCanFaceThickness/2.0 + fGermaniumDistFromCanFace + fShift + fAppliedBackShift + distFromOriginDet, and theta/phi taken from fCoords.

I think the Trans* functions are meant to combine the translation of the crystal/components within the detectors reference frame with a rotation by theta and phi within the world reference frame. My guess is that the problem is with these functions as the x,y,z used to describe the position of the crystals relative to an origin "distFromOriginDet" away from the detector looks reasonable to me and is independent of the choice of theta and phi. Maybe these functions are only right for the specific values of theta and phi that we use in the default arrangement (theta= 45, 90, or 135 degree)? At 45 degree sin and cosine are the same, and at 135 they only differ by the sign.

**Comment by VinzenzBildstein** [comment]
Okay, I think I figured out what is going on. The way the construction is set up, you can only place GRIFFIN detectors in such a way that they face the origin. Meaning you will need to set the pitch/beta to be theta - 90 degree. If you do that, you can set theta to any value you want, and both the crystals and the can/cryostat  are place correctly.
What values did you use for the fCoords of the detector you showed @scshadrick?

**Comment by scshadrick** [comment]
Sorry for the wait.  The full system of coordinates is as follows:
I used detectors 1,3,6,7,10, and 11.
    // theta
    this->coords[0][0] 	= 45.0;
    this->coords[1][0] 	= 55.0;
    this->coords[2][0] 	= 45.0;
    this->coords[3][0] 	= 55.0;
    this->coords[4][0] 	= 90.0;
    this->coords[5][0] 	= 90.0;
    this->coords[6][0] 	= 90.0;
    this->coords[7][0] 	= 90.0;
    this->coords[8][0] 	= 90.0;
    this->coords[9][0] 	= 90.0;
    this->coords[10][0] 	= 90.0;
    this->coords[11][0] 	= 90.0;
    this->coords[12][0] 	= 135.0;
    this->coords[13][0] 	= 135.0;
    this->coords[14][0] 	= 135.0;
    this->coords[15][0] 	= 135.0;
    // phi
    this->coords[0][1] 	= 67.5;
    this->coords[1][1] 	= 165.5;
    this->coords[2][1] 	= 247.5;
    this->coords[3][1] 	= 345.5;
    this->coords[4][1] 	= 22.5;
    this->coords[5][1] 	= 67.5;
    this->coords[6][1] 	= 110.5;
    this->coords[7][1] 	= 159.5;
    this->coords[8][1] 	= 202.5;
    this->coords[9][1] 	= 247.5;
    this->coords[10][1] 	= 290.5;
    this->coords[11][1] 	= 339.5;
    this->coords[12][1] 	= 67.5;
    this->coords[13][1] 	= 157.5;
    this->coords[14][1] 	= 247.5;
    this->coords[15][1] 	= 337.5;
    // yaw (alpha)
    this->coords[0][2] 	= 0.0;
    this->coords[1][2] 	= 0.0;
    this->coords[2][2] 	= 0.0;
    this->coords[3][2] 	= 0.0;
    this->coords[4][2] 	= 0.0;
    this->coords[5][2] 	= 0.0;
    this->coords[6][2] 	= 0.0;
    this->coords[7][2] 	= 0.0;
    this->coords[8][2] 	= 0.0;
    this->coords[9][2] 	= 0.0;
    this->coords[10][2] 	= 0.0;
    this->coords[11][2] 	= 0.0;
    this->coords[12][2] 	= 0.0;
    this->coords[13][2] 	= 0.0;
    this->coords[14][2] 	= 0.0;
    this->coords[15][2] 	= 0.0;
    // pitch (beta)
    this->coords[0][3] 	= 0.0;
    this->coords[1][3] 	= -20.0;
    this->coords[2][3] 	= 0.0;
    this->coords[3][3] 	= -20.0;
    this->coords[4][3] 	= 0.0;
    this->coords[5][3] 	= 0.0;
    this->coords[6][3] 	= 0.0;
    this->coords[7][3] 	= 0.0;
    this->coords[8][3] 	= 0.0;
    this->coords[9][3] 	= 0.0;
    this->coords[10][3] 	= 0.0;
    this->coords[11][3] 	= 0.0;
    this->coords[12][3] 	= 45.0;
    this->coords[13][3] 	= 45.0;
    this->coords[14][3] 	= 45.0;
    this->coords[15][3] 	= 45.0;
    // roll (gamma)
    this->coords[0][4] 	= 67.5;
    this->coords[1][4] 	= 157.5;
    this->coords[2][4] 	= 247.5;
    this->coords[3][4] 	= 337.5;
    this->coords[4][4] 	= 22.5;
    this->coords[5][4] 	= 67.5;
    this->coords[6][4] 	= 112.5;
    this->coords[7][4] 	= 157.5;
    this->coords[8][4] 	= 202.5;
    this->coords[9][4] 	= 247.5;
    this->coords[10][4] 	= 292.5;
    this->coords[11][4] 	= 337.5;
    this->coords[12][4] 	= 67.5;
    this->coords[13][4] 	= 157.5;
    this->coords[14][4] 	= 247.5;
    this->coords[15][4] 	= 337.5;

**Comment by VinzenzBildstein** [comment]
Okay, like I mentioned, you will need to set the pitch/beta to be theta - 90 degree. So keeping the theta you list, you would need to change the beta to
```
// pitch (beta)
this->coords[0][3] = -45.0;
this->coords[1][3] = -35.0;
this->coords[2][3] = -45.0;
this->coords[3][3] = -35.0;
this->coords[4][3] = 0.0;
this->coords[5][3] = 0.0;
this->coords[6][3] = 0.0;
this->coords[7][3] = 0.0;
this->coords[8][3] = 0.0;
this->coords[9][3] = 0.0;
this->coords[10][3] = 0.0;
this->coords[11][3] = 0.0;
this->coords[12][3] = 45.0;
this->coords[13][3] = 45.0;
this->coords[14][3] = 45.0;
this->coords[15][3] = 45.0;
```
This only affects the first four detectors, so the other twelve should already look alright?

**Comment by VinzenzBildstein** [comment]
Has this been solved @scshadrick ?

**Comment by scshadrick** [comment]
I haven't been working with this code since the beginning of January, but changing the beta to -20 was needed to get the detectors in the right position.  I'll have a look today to see if this fixes the issue.

**Comment by scshadrick** [comment]
So I've had a look.  The coloured bits seem to be in the right place in relation to the shell, but the geometry is now off and there is overlap in the GRIFFIN detectors I am using. 
![4vinzenzdragon](https://user-images.githubusercontent.com/20971746/53448237-26e67c00-39d4-11e9-8690-5295d07d678a.png)

**Comment by VinzenzBildstein** [comment]
Yes, I think this is because the positions you're putting the detectors in requires them to not all face the target (origin) in order not to overlap? As mentioned above, the simulation can only arrange the detectors so that they do face the target/origin.
To change this would require changing the way the detector assembly is built up.

**Comment by scshadrick** [comment]
The top two detectors need to be off-shell for the DRAGON arrangement to increase the solid angle coverage.  How much of a pain would it be to change the way it's built to necessitate off-shell detectors?

**Comment by VinzenzBildstein** [comment]
I'm not sure how difficult this would be. I think the issue is that right now the detector is assembled using all the given coordinates and then translated and rotated by these same coordinates to get it into the right spot.
This would need to be changed so that without and translation or rotation the assembly is in the right position (using hard-coded translation and rotations for the parts). Then it could be translated and rotated into any position without any restrictions on the direction it's facing.
